### ISSUE TITLE
Specify to which person a delivery script will send the email to.

### ISSUE THREAD

**Issue by Fattigman** [ctg-lund/singleCellWorkflows#38]
## Problem
The email script today is hardcoded to be delivered to myself. This is suboptimal if someone else at CTG want to run the pipeline, it will still be delivered to me.

## Suggestion
Add a config parameter that is tells the script which person the data is going to get delivered to.

## Can be closed when:
- [x] Nextflow config parameter of who is getting the delivery email

## Note
This is still a temporary solution, as the final solution will be a delivery routine by Yggdrasil or another delivery method.

**Comment by chaetognatha** [comment]
Didn't we get an email that points to the whole bioinformatics team?

**Comment by Fattigman** [comment]
When has it ever done that?

**Comment by chaetognatha** [comment]
Because an alias that points to all of us would be a great default

**Comment by chaetognatha** [comment]
So I edited the delivery script that is under Yggdrasil/bin/delivery.sh
This is how it works now:
```console
-rw-rw-r-- 1 mattis ls-grupp4 30 Aug 24 14:00 10.txt

mattis@ls4-fe1 /projects/fs1/shared/Test_Jobs/Tiny_Test $
cat *
Thu Aug 24 14:00:59 CEST 2023
Thu Aug 24 14:00:59 CEST 2023
Thu Aug 24 14:00:59 CEST 2023
Thu Aug 24 14:00:59 CEST 2023
Thu Aug 24 14:00:59 CEST 2023
Thu Aug 24 14:00:59 CEST 2023
Thu Aug 24 14:00:59 CEST 2023
Thu Aug 24 14:00:59 CEST 2023
Thu Aug 24 14:00:59 CEST 2023
Thu Aug 24 14:00:59 CEST 2023

mattis@ls4-fe1 /projects/fs1/shared/Test_Jobs/Tiny_Test $
bash ~/Shared/Yggdrasil/bin/delivery.sh 
Error: Missing required arguments. Specify at least -d directory/path and p YYYY_XXX

mattis@ls4-fe1 /projects/fs1/shared/Test_Jobs/Tiny_Test $
bash ~/Shared/Yggdrasil/bin/delivery.sh -d /projects/fs1/shared/Test_Jobs/Tiny_Test
Error: Missing required arguments. Specify at least -d directory/path and p YYYY_XXX

mattis@ls4-fe1 /projects/fs1/shared/Test_Jobs/Tiny_Test $
bash ~/Shared/Yggdrasil/bin/delivery.sh -d /projects/fs1/shared/Test_Jobs/Tiny_Test -p Tiny_Test
Error: Invalid project ID argument.

mattis@ls4-fe1 /projects/fs1/shared/Test_Jobs/Tiny_Test $
bash ~/Shared/Yggdrasil/bin/delivery.sh -d /projects/fs1/shared/Test_Jobs/Tiny_Test -p 2021_001 -u mattis
The directory is: /projects/fs1/shared/Test_Jobs/Tiny_Test
The project ID is: 2021_001
The LFS_USER is: mattis
The email is: cglu.bioinformatics@scilifelab.se
Adding user `2021_001' to group `ssh_users' ...
Adding user 2021_001 to group ssh_users
Done.
sending incremental file list
Tiny_Test/
Tiny_Test/1.txt
             30 100%    0.00kB/s    0:00:00 (xfr#1, to-chk=9/11)
Tiny_Test/10.txt
             30 100%   29.30kB/s    0:00:00 (xfr#2, to-chk=8/11)
Tiny_Test/2.txt
             30 100%   29.30kB/s    0:00:00 (xfr#3, to-chk=7/11)
Tiny_Test/3.txt
             30 100%   29.30kB/s    0:00:00 (xfr#4, to-chk=6/11)
Tiny_Test/4.txt
             30 100%   29.30kB/s    0:00:00 (xfr#5, to-chk=5/11)
Tiny_Test/5.txt
             30 100%   29.30kB/s    0:00:00 (xfr#6, to-chk=4/11)
Tiny_Test/6.txt
             30 100%   14.65kB/s    0:00:00 (xfr#7, to-chk=3/11)
Tiny_Test/7.txt
             30 100%   14.65kB/s    0:00:00 (xfr#8, to-chk=2/11)
Tiny_Test/8.txt
             30 100%   14.65kB/s    0:00:00 (xfr#9, to-chk=1/11)
Tiny_Test/9.txt
             30 100%    9.77kB/s    0:00:00 (xfr#10, to-chk=0/11)

sent 961 bytes  received 210 bytes  2.34K bytes/sec
total size is 300  speedup is 0.26

mattis@ls4-fe1 /projects/fs1/shared/Test_Jobs/Tiny_Test $
bash ~/Shared/Yggdrasil/bin/delivery.sh -d /projects/fs1/shared/Test_Jobs/Tiny_Test -p 2021_002 -e cglu.bioinformatics@scilifelab.se
The directory is: /projects/fs1/shared/Test_Jobs/Tiny_Test
The project ID is: 2021_002
The LFS_USER is: mattis
The email is: cglu.bioinformatics@scilifelab.se
Adding user `2021_002' to group `ssh_users' ...
Adding user 2021_002 to group ssh_users
Done.
sending incremental file list
Tiny_Test/
Tiny_Test/1.txt
             30 100%    0.00kB/s    0:00:00 (xfr#1, to-chk=9/11)
Tiny_Test/10.txt
             30 100%   29.30kB/s    0:00:00 (xfr#2, to-chk=8/11)
Tiny_Test/2.txt
             30 100%   29.30kB/s    0:00:00 (xfr#3, to-chk=7/11)
Tiny_Test/3.txt
             30 100%   29.30kB/s    0:00:00 (xfr#4, to-chk=6/11)
Tiny_Test/4.txt
             30 100%   29.30kB/s    0:00:00 (xfr#5, to-chk=5/11)
Tiny_Test/5.txt
             30 100%   29.30kB/s    0:00:00 (xfr#6, to-chk=4/11)
Tiny_Test/6.txt
             30 100%   29.30kB/s    0:00:00 (xfr#7, to-chk=3/11)
Tiny_Test/7.txt
             30 100%   14.65kB/s    0:00:00 (xfr#8, to-chk=2/11)
Tiny_Test/8.txt
             30 100%   14.65kB/s    0:00:00 (xfr#9, to-chk=1/11)
Tiny_Test/9.txt
             30 100%   14.65kB/s    0:00:00 (xfr#10, to-chk=0/11)

sent 961 bytes  received 210 bytes  2.34K bytes/sec
total size is 300  speedup is 0.26

mattis@ls4-fe1 /projects/fs1/shared/Test_Jobs/Tiny_Test $
bash ~/Shared/Yggdrasil/bin/delivery.sh -d /projects/fs1/shared/Test_Jobs/Tiny_Test -p 2021_002 -e mattis.knulst@med.lu.se
The directory is: /projects/fs1/shared/Test_Jobs/Tiny_Test
The project ID is: 2021_002
The LFS_USER is: mattis
The email is: mattis.knulst@med.lu.se
The user `2021_002' is already a member of `ssh_users'.
sending incremental file list
Tiny_Test/1.txt
             30 100%    0.00kB/s    0:00:00 (xfr#1, to-chk=9/11)
Tiny_Test/10.txt
             30 100%   29.30kB/s    0:00:00 (xfr#2, to-chk=8/11)
Tiny_Test/2.txt
             30 100%   29.30kB/s    0:00:00 (xfr#3, to-chk=7/11)
Tiny_Test/3.txt
             30 100%   29.30kB/s    0:00:00 (xfr#4, to-chk=6/11)
Tiny_Test/4.txt
             30 100%   29.30kB/s    0:00:00 (xfr#5, to-chk=5/11)
Tiny_Test/5.txt
             30 100%   29.30kB/s    0:00:00 (xfr#6, to-chk=4/11)
Tiny_Test/6.txt
             30 100%   29.30kB/s    0:00:00 (xfr#7, to-chk=3/11)
Tiny_Test/7.txt
             30 100%   14.65kB/s    0:00:00 (xfr#8, to-chk=2/11)
Tiny_Test/8.txt
             30 100%   14.65kB/s    0:00:00 (xfr#9, to-chk=1/11)
Tiny_Test/9.txt
             30 100%   14.65kB/s    0:00:00 (xfr#10, to-chk=0/11)

sent 658 bytes  received 267 bytes  1.85K bytes/sec
total size is 300  speedup is 0.32
```

**Comment by chaetognatha** [comment]
TL;DR validation seems to be working and you can specify your own email and that works, unfortunately the default email whether you specify it directly or default to it doesn't seem to work, but that is an issue I need to look more closely at, for now this delivery script has been generalized so we can use it directly

**Comment by Fattigman** [comment]
Cool, is it on the master branch on Yggdrasil or?

There are some more disussions regarding implementation of delivery, but I think that is a discussion for a separate issue.

**Comment by chaetognatha** [comment]
Yes on the master branch!

Agreed, I think at some point the delivery script should go in its own repo, just as the demux script that Henning is writing also should have its own repo so that we can isolate them better!

**Comment by Fattigman** [comment]
Cool, I will start on the changes to this code base to call on the new delivery script.

**Comment by Fattigman** [comment]
Fixed in https://github.com/ctg-lund/singleCellWorkflows/commit/fca3090c00378d14db2d3218f387304f553af57c

**Comment by chaetognatha** [comment]
👀
